### ISSUE TITLE
Show offset text on subplots after sharing axis

### ISSUE THREAD

**Issue by hamzamerzic** [matplotlib/matplotlib#15225]
I'm trying to display xticks in parts of the subplot when sharing x axis as shown in the image below.

![image](https://user-images.githubusercontent.com/10846014/64490694-a1336b00-d257-11e9-9bd9-f21049c789da.png)

To be able to display xticks for subplots not at the bottom, I used `ax.xaxis.set_tick_params(labelbottom=True)`, which works well unless the x-axis limit is not too big. As you can see in the image above `1e8` is missing. I'm not sure how that part of the ticks is called, so my search results ended up poorly. Any help on how to include this within the axis ticks or what to search for is appreciated. Thank you!

The code to reproduce:
```python
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns
sns.set_style('whitegrid')

sub_w, sub_h = 6, 4
# Figure layout. Want to plot three subplots, but need to allocate a grid of 4.
fig, axg = plt.subplots(2, 2, sharex=True, sharey=False, 
                        figsize=(sub_w * 2, sub_h * 2))
x = 100000 * np.arange(1000, dtype=np.float32)
y = np.sin(x / 10000000)

handles = []
num_plots = 3
num_lines_per_plot = 2
colors = sns.color_palette('husl', n_colors=num_lines_per_plot)
# The following line makes sure xticks are shown on the top right subplot.
axg.flat[1].xaxis.set_tick_params(labelbottom=True)
for i in range(num_plots):
  ax = axg.flat[i]
  for j in range(num_lines_per_plot):
    ax.plot(x, y * (i + j + 1), color=colors[j])

plt.tight_layout()

for aa in axg.flat[3:]:
  aa.set_visible(False)
```

**Comment by jklymak** [comment]
I don't know how `labelbottom` works, but it appears to not be making the offset text visible?  Checkout `ax.get_xaxis().get_offset_text()` for a handle to that.

**Comment by hamzamerzic** [comment]
Thank you for the pointer! Adding `axg.flat[1].xaxis.offsetText.set_visible(True)` solves my issue.

**Comment by ImportanceOfBeingErnest** [comment]
Reopening because clearly this should fail.

**Comment by timhoffm** [comment]
Not quite clear on the scope of `labelbottom`, but should it affect the `offsetText` as well?

**Comment by jklymak** [comment]
I’d think so.  It’s impossible to understand the tick labels without it. Just seems an oversight that it wasn’t turned on.

**Comment by ImportanceOfBeingErnest** [comment]
The color is special cased 

https://github.com/matplotlib/matplotlib/blob/e69699f9767799607761b2886668eaf736cc81e5/lib/matplotlib/axis.py#L915-L918

but visibility is not. That's probably also because it's hard to define. If labelbottom=False and labeltop=False, it should be invisible. But if only one of them is False, it might still need to be visible. Also the user might have set its visibility off, in which case it should probably stay off.

**Comment by timhoffm** [comment]
I think `set_tick_params(labelbottom=True)` should re-enable the `offsetText` no matter if the user has set it explicitly to off. If the user want's to hide it, that should be done after messing with the tick params.

**Comment by RiazCharania** [comment]
Is anyone else currently working on this? If not I'll give it a try.

**Comment by timhoffm** [comment]
You are welcome to.

**Comment by Anurudran** [comment]
We decided to not work on this issue.

**Comment by QuLogic** [comment]
This was fixed by #16896.
